### PR TITLE
Contract definitions page

### LINKED ISSUE #10: Contract Definitions Page

### LINKED ISSUE #16: Contract Definitions Page
Feature #14

## Description

The current "Contract Definitions" page supports creating new contract definitions by introducing the "asset property", "operator and "value" triplet as text. This is not user-friendly and must be improved.

## Acceptance Criteria
- [x] Improve UI to allow selection from available assets (multi-selection)

## Tasks
- [x] Rename page from "Contract Definition Editor" to "Contract Definitions"
- [x] Improve UI to allow selection from available assets (multi-selection)

### PR DESCRIPTION
**Pull request by marcgs**
closes agera-edc/DataDashboardFork#10 

Contract definition page support multi selection of existing assets.

See previous PR for renamings: agera-edc/EDCDataDashboard#23 

Running sample: http://uitest469-company2-mvd.northeurope.azurecontainer.io

### CODE REVIEW

File: src/modules/edc-demo/components/contract-definition-editor-dialog/contract-definition-editor-dialog.component.html [hunk 1]
```diff
@@ -23,43 +23,16 @@
...
-            </mat-card-actions>
-
-        </mat-card-content>
-    </mat-card>
-    <div>&nbsp;</div>
+        <mat-form-field class="form-field form-field-assets" id="form-field-contract-assets">
+          <mat-label>Assets</mat-label>
+          <mat-select [(ngModel)]="assets" multiple>
+            <mat-option *ngFor="let asset of availableAssets" [value]="asset">
+              {{asset.properties["asset:prop:id"]}}
```
Comment (paullatzelsperger) [inline src/modules/edc-demo/components/contract-definition-editor-dialog/contract-definition-editor-dialog.component.html:31]: maybe refactor into a method on the asset object. Check the internal model class `Asset`, maybe it's better to map the `AssetDto` into an `Asset` and use that.

**Comment by paullatzelsperger** [comment]
I think the `Edit` button of the ContractDefiniiton Viewer should go away. There is no way to "edit" a CD in the DM API, only to create and delete.

File: src/modules/edc-demo/components/contract-definition-editor-dialog/contract-definition-editor-dialog.component.html (hunk 1 above)
Comment (marcgs) [reply src/modules/edc-demo/components/contract-definition-editor-dialog/contract-definition-editor-dialog.component.html:31]: > I think the `Edit` button of the ContractDefiniiton Viewer should go away. There is no way to "edit" a CD in the DM API, only to create and delete.

Currently the `edit` button does exactly this: remove and then recreate. Should I delete it?

**Comment by marcgs** [comment]
> I think the `Edit` button of the ContractDefiniiton Viewer should go away. There is no way to "edit" a CD in the DM API, only to create and delete.

Currently the `edit` button does exactly this: remove and then recreate. Should I delete it?

**Review by chrislomonico — APPROVED** [review]

**Comment by marcgs** [comment]
> > I think the `Edit` button of the ContractDefiniiton Viewer should go away. There is no way to "edit" a CD in the DM API, only to create and delete.
> 
> Currently the `edit` button does exactly this: remove and then recreate. Should I delete it?

Actually you are right, in this case the `edit` was not recreating the Contract Definition. I will delete the button.

**Pull request merged**